src/components/course/CourseSidebarPrice.jsx

```diff
@@ -3,17 +3,19 @@ import Skeleton from 'react-loading-skeleton';
 import { AppContext } from '@edx/frontend-platform/react';
 
 import { CourseContext } from './CourseContextProvider';
+import { UserSubsidyContext } from '../enterprise-user-subsidy/UserSubsidy';
+import { numberWithPrecision, hasLicenseSubsidy } from './data/utils';
 
 import {
   useCoursePriceForUserSubsidy,
 } from './data/hooks';
-import { hasLicenseSubsidy } from './data/utils';
 
 const CourseSidebarPrice = () => {
   const { state } = useContext(CourseContext);
-  const { activeCourseRun, userSubsidy } = state;
+  const { activeCourseRun, userSubsidy, catalog: { catalogList } } = state;
```

Comment:
Not really about this ticket, more for me if you have time to explain: why does our design have a userSubsidy state AND userSubsidy context? Isn't the point of a context to move something out of a state?

Reply:
I can add some historical context here. The user subsidy context is largely keeping track of all the subsidies a user has assigned to them (i.e., license, offers) but it doesn't take into account _which_ of those subsidy, if any, apply to the course that the user is viewing. For example, a user's license might be associated with a catalog for which the course is not part of so we'd have to fallback to using an offer, assuming it applies to the course, etc. which was the course context `userSubsidy` was meant for.

I'm not sure if the current implementation fully goes by that design but that was the original idea.

Totally agree the naming of `userSubsidy` in the course context could be clearer, maybe `userSubsidyApplicableToCourse`?

Reply:
good question. first off, I think the name state here may be misleading. It really is just the course context. Another, in my upcoming refactor (that got parked) I am setting up a new userSubsidy hook that will manage all subsidy related stuff leading to less raw context use in code like this. but when doing that I can pay special attention to whether we can further streamline the use of the context

does this help?

Reply:
ah two points:

[1] That `state` var which I mentioned:
I did not clarify much.. that `state` is a pattern that comes out the user of the userReducer hook 
Typically looks like this
```
  const { state, dispatch } = useContext(MyContext);
```
for that part I am going with a small rename (since I must use 'state' due to the reducer.. ) 

```
  const { state: courseInfo } = useContext(CourseContext);
```

[2] As for the userSubsidy, yes that confuses matters a bit with the main subsidy context so probably best renamed: https://github.com/edx/frontend-app-learner-portal-enterprise/blob/3b1992b0b366177922afb91ba9cd87b958d6758d/src/components/enterprise-user-subsidy/UserSubsidy.jsx#L10